src/lib/components/TestimonialCarousel.svelte

```diff
@@ -0,0 +1,226 @@
+      rootMargin: "0px",
+      threshold: [0.25, 0.75, 1],
+    };
+
+    const observer = new IntersectionObserver((observerEntries) => {
+      for (const entry of observerEntries) {
+        if (entry.intersectionRatio >= 0.75) {
+          viewingIndex = testimonialElements.findIndex(
+            (elem) => entry.target === elem
+          );
+        }
+      }
+    }, observerOptions);
+
+    for (const element of testimonialElements) {
+      observer.observe(element);
+    }
+
+    return () => observer.disconnect();
+  });
+
+  async function scroll(index: number) {
+    if (index < 0) {
+      index = testimonialElements.length + index;
+    }
+
+    testimonialElements[index % testimonialElements.length].scrollIntoView({
+      behavior: "smooth",
+      inline: "center",
+      block: "nearest",
+    });
+  }
+</script>
+
+<div class="wrapper">
+  <div class="scroller">
+    <div class="column-center" bind:this={scrollerChild}>
+      <div class="carousel">
+        {#each testimonials as testimonial, i (testimonial)}
+          <div class="testimonial-wrapper" bind:this={testimonialElements[i]}>
+            <Testimonial
+              quote={testimonial.content}
+              name={testimonial.sourceName}
+              desc={testimonial.sourceDescription}
+              imageSrc={testimonial.sourceImage?.src}
+            />
+          </div>
+        {/each}
+      </div>
+    </div>
+    {#if testimonials.length > 1}
+      <div class="guide">
+        {#each testimonials as _, i}
+          <button
+            aria-label="See testimonial {i + 1}"
+            class:active={viewingIndex === i}
+            on:click={() => scroll(i)}
+          />
+        {/each}
+      </div>
+
+      <button
+        class="page-button prev"
+        on:click={() => scroll(viewingIndex - 1)}
+      >
+        <Icon icon="left-arrow" width="2em" height="2em" />
+      </button>
+      <button
+        class="page-button next"
+        on:click={() => scroll(viewingIndex + 1)}
+      >
+        <Icon icon="right-arrow" width="2em" height="2em" />
+      </button>
+    {/if}
+  </div>
+</div>
+
+<style>
+  .wrapper {
+    position: relative;
+  }
+
+  .scroller {
+    overflow-x: auto;
+    scroll-snap-type: x mandatory;
+  }
+
+  .scroller::-webkit-scrollbar {
+    display: none;
+  }
+
+  .wrapper::before,
+  .wrapper::after {
+    content: "";
+    position: absolute;
+    top: 0;
+    left: calc(calc(100% - var(--content-width)) / 2);
+    height: 100%;
+    width: calc(
+      calc(calc(var(--content-width) - var(--long-form-width)) / 2) * 0.75
+    );
+    --gradient-colors: var(--testimonial-background, var(--gray-lighter)),
+      transparent;
+    background: linear-gradient(90deg, var(--gradient-colors));
+    z-index: 20;
+  }
+
+  .wrapper::after {
+    left: unset;
+    right: calc(calc(100% - var(--content-width)) / 2);
+    background: linear-gradient(-90deg, var(--gradient-colors));
+  }
+
+  :where(.page-button) {
```

Comment:
These buttons might look better in dark blue since we only use black/grey for text.